File: iOS/TaskBridge/Repository/Impl/TasksRepositoryImpl.swift
import Foundation
import TaskBridgeCore

@MainActor
class TasksRepositoryImpl: TasksRepository {
    private let interactor: TasksInteractor
    
    init(interactor: TasksInteractor) {
        self.interactor = interactor
    }
    
    var tasksState: AsyncStream<TasksState> {
        return createStream(for: interactor.tasksState)
    }
    
    func loadTasks() async throws {
        try await interactor.loadTasks()
    }
    
    func createTask(_ task: TaskItem) async throws {
        try await interactor.createTask(task: task)
    }
    
    func replaceTask(_ task: TaskItem) async throws {
        try await interactor.replaceTask(task: task)
    }
    
    func deleteTaskTree(_ taskId: TaskId) async throws {
        try await interactor.deleteTaskTree(taskId: taskId)
    }
}

File: Android/src/androidMain/kotlin/com/taskbridge/android/repository/impl/TasksRepositoryImpl.kt
package com.taskbridge.android.repository.impl

import com.taskbridge.android.repository.TasksRepository
import com.taskbridge.core.interactors.tasks.TasksInteractor
import com.taskbridge.core.models.tasks.TaskId
import com.taskbridge.core.models.tasks.TaskItem
import com.taskbridge.core.usecases.tasks.TasksState
import kotlinx.coroutines.flow.Flow

class TasksRepositoryImpl(
    private val interactor: TasksInteractor
) : TasksRepository {
    override val tasksState: Flow<TasksState> = interactor.tasksState

    override suspend fun loadTasks() {
        interactor.loadTasks()
    }

    override suspend fun createTask(task: TaskItem) {
        interactor.createTask(task)
    }

    override suspend fun replaceTask(task: TaskItem) {
        interactor.replaceTask(task)
    }

    override suspend fun deleteTaskTree(taskId: TaskId) {
        interactor.deleteTaskTree(taskId)
    }
}
